### PR TITLE
Voucher link adjust stage

### PR DESCRIPTION
**Pull request by prateek-giddh**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts [hunk 1]
```diff
@@ -1456,12 +1613,12 @@ export class UpdateLedgerEntryPanelComponent implements OnInit, AfterViewInit, O
      * @memberof UpdateLedgerEntryPanelComponent
      */
     public adjustedInvoiceAmountChange(): void {
-        if (this.vm.selectedLedger && this.vm.selectedLedger.invoiceAdvanceReceiptAdjustment && this.vm.selectedLedger.invoiceAdvanceReceiptAdjustment.adjustedInvoices) {
+        if (this.vm.selectedLedger && this.vm.selectedLedger.voucherAdjustments && this.vm.selectedLedger.voucherAdjustments.adjustments) {
             let totalAmount: number = 0;
-            this.vm.selectedLedger.invoiceAdvanceReceiptAdjustment.adjustedInvoices.forEach(item => {
-                totalAmount = Number(totalAmount) + Number(item.adjustedAmount.amountForAccount);
+            this.vm.selectedLedger.voucherAdjustments.adjustments.forEach(item => {
+                totalAmount = Number(totalAmount) + Number(item.adjustmentAmount.amountForAccount);
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts:1619]: Add nullity chekck for adjustmentAmount

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts [hunk 2]
```diff
@@ -1484,10 +1641,10 @@ export class UpdateLedgerEntryPanelComponent implements OnInit, AfterViewInit, O
     * @memberof UpdateLedgerEntryPanelComponent
     */
     public adjustedReceiptsAmountChange(): void {
-        if (this.vm.selectedLedger && this.vm.selectedLedger.advanceReceiptAdjustment && this.vm.selectedLedger.advanceReceiptAdjustment.adjustments) {
+        if (this.vm.selectedLedger && this.vm.selectedLedger.voucherAdjustments && this.vm.selectedLedger.voucherAdjustments.adjustments) {
             let totalAmount: number = 0;
-            this.vm.selectedLedger.advanceReceiptAdjustment.adjustments.forEach(item => {
-                totalAmount = Number(totalAmount) + Number(item.dueAmount.amountForAccount);
+            this.vm.selectedLedger.voucherAdjustments.adjustments.forEach(item => {
+                totalAmount = Number(totalAmount) + Number(item.adjustmentAmount.amountForAccount);
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts:1647]: Add nullity check for adjustmentAmount

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts [hunk 3]
```diff
@@ -1551,4 +1709,133 @@ export class UpdateLedgerEntryPanelComponent implements OnInit, AfterViewInit, O
...
     }
+
+    /**
+     * Payment adjustment handler
+     *
+     * @param {{ adjustVoucherData: VoucherAdjustments, adjustPaymentData: AdjustAdvancePaymentModal}} event Adjustment handler
+     * @memberof UpdateLedgerEntryPanelComponent
+     */
+    public getAdjustedPaymentData(event: { adjustVoucherData: VoucherAdjustments, adjustPaymentData: AdjustAdvancePaymentModal}): void {
+        console.log(this.vm.selectedLedger);
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts:1720]: Remove console

File: apps/web-giddh/src/app/ledger/ledger.component.ts [hunk 4]
```diff
@@ -886,6 +886,33 @@ export class LedgerComponent implements OnInit, OnDestroy {
...
+     * @memberof LedgerComponent
+     */
+    public getInvoiceListsForCreditNote(voucherType: string): void {
+        if (voucherType && this.selectedTxnAccUniqueName && this.accountUniquename) {
+            const request = {
+                accountUniqueNames: [this.selectedTxnAccUniqueName, this.accountUniquename],
+                voucherType
+            };
+            let date;
+            if (typeof this.lc.blankLedger.entryDate === 'string') {
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/ledger/ledger.component.ts:902]: nullity check for 'entryDate'

File: apps/web-giddh/src/app/models/api-models/recipt.ts [hunk 5]
```diff
@@ -46,7 +46,8 @@ export interface ReceiptItem {
 	dueDate: string;
 	isSelected?: boolean;
 	cashInvoice: boolean;
-	accountCurrencySymbol?: string;
+    accountCurrencySymbol?: string;
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/models/api-models/recipt.ts:49]: run formatter

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts [hunk 6]
```diff
@@ -1639,7 +1729,8 @@ export class ProformaInvoiceComponent implements OnInit, OnDestroy, AfterViewIni
         this.isGenDtlCollapsed = true;
         this.isMlngAddrCollapsed = true;
         this.isOthrDtlCollapsed = false;
-        this.forceClear$ = observableOf({status: true});
+        this.forceClear$ = observableOf({ status: true });
+        this.invoiceForceClearReactive$ = observableOf({ status: true });
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts:1733]: Add doc

File: apps/web-giddh/src/app/shared/advance-receipt-adjustment/advance-receipt-adjustment.component.ts [hunk 7]
```diff
@@ -106,9 +119,55 @@ export class AdvanceReceiptAdjustmentComponent implements OnInit {
             this.invoiceFormDetails.voucherDetails.tdsTotal = this.invoiceFormDetails.voucherDetails.tdsTotal || 0;
             this.assignVoucherDetails();
         }
-        this.getAllAdvanceReceipts();
+        console.log('Invoice form details: ', this.invoiceFormDetails);
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/shared/advance-receipt-adjustment/advance-receipt-adjustment.component.ts:122]: remove console

File: apps/web-giddh/src/app/shared/advance-receipt-adjustment/advance-receipt-adjustment.component.ts [hunk 8]
```diff
@@ -106,9 +119,55 @@ export class AdvanceReceiptAdjustmentComponent implements OnInit {
...
+            } else {
+                // A ledger entry is updated
+                requestObject = {
+                    accountUniqueNames: [...customerUniqueName, this.invoiceFormDetails.activeAccountUniqueName],
+                    voucherType,
+                    subVoucher: this.adjustedVoucherType === AdjustedVoucherType.AdvanceReceipt ? SubVoucher.AdvanceReceipt : undefined
+                }
+            }
+            this.salesService.getInvoiceList(requestObject, this.invoiceFormDetails.voucherDetails.voucherDate).subscribe((response) => {
+                console.log('Response received: ', response);
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/shared/advance-receipt-adjustment/advance-receipt-adjustment.component.ts:144]: remove console

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts [hunk 9]
```diff
@@ -1147,22 +1190,116 @@ export class UpdateLedgerEntryPanelComponent implements OnInit, AfterViewInit, O
...
+                this.isAdjustVoucherSelected = false;
+            }
+            return;
+        }
+        if (this.vm.selectedLedger.voucherAdjustments && !this.vm.selectedLedger.voucherAdjustments.adjustments ) {
+            this.vm.selectedLedger.voucherAdjustments.adjustments = [];
+        }
+        if ((this.isAdjustAdvanceReceiptSelected || this.isAdjustReceiptSelected || this.isAdjustVoucherSelected) && this.vm.selectedLedger.voucherAdjustments && (!this.vm.selectedLedger.voucherAdjustments.adjustments.length || isUpdateMode)) {
+            this.prepareAdjustVoucherConfiguration();
+            console.log(this.vm.selectedLedger);
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts:1245]: remove console

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts [hunk 10]
```diff
@@ -1551,4 +1709,133 @@ export class UpdateLedgerEntryPanelComponent implements OnInit, AfterViewInit, O
...
+                });
+
+                this.vm.selectedLedger.voucherAdjustments = {
+                    adjustments,
+                    totalAdjustmentAmount: event.adjustPaymentData.totalAdjustedAmount,
+                    tdsTaxUniqueName: null,
+                    tdsAmount: null,
+                    description: null
+                };
+                console.log(this.vm.selectedLedger);
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts:1736]: remove console

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts [hunk 11]
```diff
@@ -1551,4 +1709,133 @@ export class UpdateLedgerEntryPanelComponent implements OnInit, AfterViewInit, O
...
+     * Close voucher adjustment modal handler
+     *
+     * @param {{ adjustVoucherData: VoucherAdjustments, adjustPaymentData: AdjustAdvancePaymentModal}} event Close event
+     * @memberof UpdateLedgerEntryPanelComponent
+     */
+    public closeAdjustmentModal(event: { adjustVoucherData: VoucherAdjustments, adjustPaymentData: AdjustAdvancePaymentModal}): void {
+        if (this.vm.selectedLedger.voucherAdjustments && this.vm.selectedLedger.voucherAdjustments.adjustments && !this.vm.selectedLedger.voucherAdjustments.adjustments.length) {
+            // No adjustments done clear the adjustment checkbox
+            this.isAdjustReceiptSelected = false;
+            this.isAdjustVoucherSelected = false
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts:1760]: add semicolon

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts [hunk 12]
```diff
@@ -1551,4 +1709,133 @@ export class UpdateLedgerEntryPanelComponent implements OnInit, AfterViewInit, O
...
+                customerUniquename: customerUniqueName,
+                totalTaxableValue: this.vm.selectedLedger.actualAmount,
+                subTotal: this.vm.selectedLedger.total.amount,
+            },
+            accountDetails: {
+                currencySymbol: this.profileObj ? this.profileObj.baseCurrencySymbol : ''
+            },
+            activeAccountUniqueName: this.activeAccount.uniqueName
+        };
+        console.log('Reached', this.vm.selectedLedger);
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts:1802]: remove console

File: apps/web-giddh/src/app/shared/advance-receipt-adjustment/advance-receipt-adjustment.component.ts [hunk 13]
```diff
@@ -106,9 +119,55 @@ export class AdvanceReceiptAdjustmentComponent implements OnInit {
...
+                if (response && response.body) {
+                    this.allAdvanceReceiptResponse = response.body.results
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/advance-receipt-adjustment/advance-receipt-adjustment.component.ts:146]: add semicolon

File: apps/web-giddh/src/app/models/api-models/recipt.ts (hunk 5 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/models/api-models/recipt.ts:49]: @rohit-walkover it is due to Github diff the file is already formatted

File: apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts (hunk 6 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/proforma-invoice/proforma-invoice.component.ts:1733]: @rohit-walkover this is written in reset method and not member declaration

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.html [hunk 14]
```diff
@@ -596,97 +705,135 @@ <h3 *ngIf="!openDropDown" (dblclick)="openHeaderDropDown()" class="d-inline-bloc
...
+                        <p *ngIf="isAdjustedAmountExcess" class="font-12 text-light">Excess
+                            amount:{{ vm.baseCurrencyDetails?.symbol}}{{adjustedExcessAmount | giddhCurrency}} <i
+                                class="icon-info pointer" tooltip="Adjustment amount is more than receipts."></i></p>
                     </div>
                     <!-- End advance receipts adjusted Invoice -->
                     <!--Adjusted advance receipts list  -->
-                    <div class="clearfix adjust-receipt font-14" *ngIf="isAdjustedWithAdvanceReceipt && vm.selectedLedger.voucherGeneratedType ==='sales'">
-                        <p class="bold">Adjusted Receipts
-                            <a href="javascript:void(0)" *ngIf="!selectedAdvanceReceiptAdjustInvoiceEditMode" (click)="openAdjustInvoiceEditMode();$event.stopPropagation();"><i class="icon-edit11 font-10"></i></a>
+                    <!-- <div class="clearfix adjust-receipt font-14" *ngIf="isAdjustedWithAdvanceReceipt && (isAdjustAdvanceReceiptSelected)">
```
[the remark this answers is not in the document]
Comment (prateek-giddh) [reply apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.html:784]: No we are keeping it and will be removed if no issues are found once this is live

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts (hunk 1 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts]: Done @rohit-walkover

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts (hunk 2 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts]: Done @rohit-walkover

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts (hunk 3 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts]: Done @rohit-walkover

File: apps/web-giddh/src/app/shared/advance-receipt-adjustment/advance-receipt-adjustment.component.ts (hunk 7 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/shared/advance-receipt-adjustment/advance-receipt-adjustment.component.ts]: Done @rohit-walkover

File: apps/web-giddh/src/app/shared/advance-receipt-adjustment/advance-receipt-adjustment.component.ts (hunk 8 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/shared/advance-receipt-adjustment/advance-receipt-adjustment.component.ts]: Done @rohit-walkover

File: apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.ts [hunk 15]
```diff
@@ -1106,12 +1176,118 @@ export class NewLedgerEntryPanelComponent implements OnInit, OnDestroy, OnChange
...
+
+    /**
+     * To make value alphanumeric
+     *
+     * @param {*} event Template ref to get value
+     * @memberof NewLedgerEntryPanelComponent
+     */
+    public allowAlphanumericChar(event: any): void {
+        if (event && event.value) {
+            this.blankLedger.passportNumber = this.generalService.allowAlphanumericChar(event.value)
```
[the remark this answers is not in the document]
Comment (prateek-giddh) [reply apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.ts]: Done

File: apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.ts [hunk 16]
```diff
@@ -1106,12 +1176,118 @@ export class NewLedgerEntryPanelComponent implements OnInit, OnDestroy, OnChange
...
+    }
+
+    /**
+     * Payment adjustment handler
+     *
+     * @param {{ adjustVoucherData: VoucherAdjustments, adjustPaymentData: AdjustAdvancePaymentModal}} event Adjustment handler
+     * @memberof NewLedgerEntryPanelComponent
+     */
+    public getAdjustedPaymentData(event: { adjustVoucherData: VoucherAdjustments, adjustPaymentData: AdjustAdvancePaymentModal}): void {
+        console.log(event);
```
[the remark this answers is not in the document]
Comment (prateek-giddh) [reply apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.ts]: Done

File: apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.ts [hunk 17]
```diff
@@ -1106,12 +1176,118 @@ export class NewLedgerEntryPanelComponent implements OnInit, OnDestroy, OnChange
...
+                // delete adjustment.balanceDue;
+            })
+            this.currentTxn.voucherAdjustments = {
+                adjustments,
+                totalAdjustmentAmount: event.adjustPaymentData.totalAdjustedAmount,
+                tdsTaxUniqueName: null,
+                tdsAmount: null,
+                description: null
+            };
+            console.log(this.currentTxn);
```
[the remark this answers is not in the document]
Comment (prateek-giddh) [reply apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.ts]: Done

File: apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.ts [hunk 18]
```diff
@@ -1168,16 +1336,40 @@ export class NewLedgerEntryPanelComponent implements OnInit, OnDestroy, OnChange
...
+                customerUniquename: this.currentTxn.selectedAccount ? this.currentTxn.selectedAccount.uniqueName : '',
+                totalTaxableValue: this.currentTxn.total,
+                subTotal: this.currentTxn.total,
+            },
+            accountDetails: {
+                currencySymbol: this.currentTxn.selectedAccount ? this.currentTxn.selectedAccount.currencySymbol : this.blankLedger.baseCurrencyToDisplay.symbol
+            },
+            activeAccountUniqueName: this.activeAccount.uniqueName
+        };
+        console.log('Reached', this.currentTxn);
```
[the remark this answers is not in the document]
Comment (prateek-giddh) [reply apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.ts]: Done

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts (hunk 9 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts]: Done

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts (hunk 10 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts]: Done

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts (hunk 11 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts]: Done

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts (hunk 12 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedgerEntryPanel.component.ts]: Done

File: apps/web-giddh/src/app/shared/advance-receipt-adjustment/advance-receipt-adjustment.component.ts (hunk 13 above)
Comment (prateek-giddh) [reply apps/web-giddh/src/app/shared/advance-receipt-adjustment/advance-receipt-adjustment.component.ts]: Done

**Pull request merged**